### ISSUE TITLE
`newline-per-chained-call`: bad indentation, and bare fn call treated as chain

### ISSUE THREAD

**Issue by OJFord** [eslint-stylistic/eslint-stylistic#265]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

With `"@stylistic/newline-per-chained-call": ["error", {ignoreChainWithDepth: 1}]`:

```ts
export const post = sqliteTable(
  'post',
  {
    id: integer('id').primaryKey(),
    name: text('name'),
  },
)
```

is reformatted to:

```ts
export const post = sqliteTable(
  'post',
  {
    id: integer('id')
.primaryKey(),
    name: text('name'),
  },
)
```

Which:
1. I would not expect anyway, since `integer('id').primaryKey()` is to me a chain of 1 - it would be fine if it were `integer.primaryKey()`, I expected that a 'chain' refers to occurrences of the `.` operator;
2. is weird indentation, not that there's really a great solution anyway

A workaround is to introduce parens:
```ts
export const post = sqliteTable(
  'post',
  {
    id: (
        integer('id')
        .primaryKey()
    ),
    name: text('name'),
  },
)
```

which is a bit verbose but less odd looking than the zero indentation above.

---

To address (1), I think this should not be treated as a >1 chain.

---

Assuming it were `integer('id').something.primaryKey()` then, to address (2), I would suggest a configurable option for what happens in object values.

```ts
/* ["error", {ignoreChainWithDepth: 1, ignoreChainInObjectValue: true}]] */

// GOOD
{
  id: foo().bar().baz(),
}

(
  foo()
  .bar()
  .baz()
)

// BAD
(
  foo().bar().baz()
)
```

then if it really does get too long and we want to force it, we can always:
```ts
// GOOD
{
  id: (
    foo()
    .bar()
    .baz()
  ),
}
```

or this could even just be new, unconfigurable, behaviour - since I'm really not sure what a sensible indentation would look like, all possibilities seem a bit weird:
```ts
{
  a: foo()
.bar()
.baz(),

  b: foo()
     .bar()
     .baz(),

  c: foo()
  .bar()
  .baz(),
}
```
or whatever else.

### Reproduction

See description, I can't work out how to use Stackblitz for it

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
